### PR TITLE
Lint away 2 TODOs from the .rubocop_todo.yml

### PR DESCRIPTION
**Pull request by olleolleolle**
This PR fixes lint issues and regenerates the TODO file.

---

Method to locate lints one at a time:

- choose a "cop" from the `.rubocop_todo.yml` file
- remove the line which includes the `.rubocop_todo.yml`
- `rubocop --only Lint/IneffectiveAccessModifier`
- fix issues and test fixes with the `--only` invocation
- `rubocop --auto-gen-config` to regenerate the `.rubocop_todo.yml` file
- `git checkout .rubocop.yml` to add back include line
- commit this change with the name of the "cop" used

### CODE REVIEW

File: lib/fakefs/dir.rb [hunk 1]
```diff
@@ -104,13 +104,13 @@ def self.glob(pattern, _flags = 0, &block)
...
       end
 
-      if pattern.is_a? Array
-        files = pattern.map do |matcher|
-          matches_for_pattern.call matcher
-        end.flatten
-      else
-        files = matches_for_pattern.call pattern
-      end
+      files = if pattern.is_a?(Array)
```
Comment (grosser) [inline lib/fakefs/dir.rb]: prefer this to stay with 2-space indent and don't waste whitespace or having to adjust whitespace when var name changes
```
files = 
  if dfsdfsfd
```

Comment (olleolleolle) [inline lib/fakefs/dir.rb]: Your answer was so terse I missed the whole point. Now taken, and implemented.

File: lib/fakefs/pathname.rb [hunk 2]
```diff
@@ -529,11 +530,11 @@ def children(with_directory = true)
         result = []
         Dir.foreach(@path) do |e|
           next if e == '.' || e == '..'
-          if with_directory
-            result << self.class.new(File.join(@path, e))
-          else
-            result << self.class.new(e)
-          end
+          result << if with_directory
```
Comment (grosser) [inline lib/fakefs/pathname.rb]: 1 more :D

Comment (olleolleolle) [inline lib/fakefs/pathname.rb]: Ah, I'm a slob. I'll fix!

**Comment by grosser** [comment]
❤️  good step in the right direction :)
I'll wait for green

**Pull request merged**